### PR TITLE
Improve handling of bitcoind-related connection/timeout issues

### PR DESCRIPTION
**Pull request by shesek**

### CODE REVIEW

**Comment by shesek** [comment]
Not sure what the actual timeout value should be, used 5 seconds as a placeholder. Thoughts?

**Comment by shesek** [comment]
Increased timeout to 10 seconds, removed the commit that added a reconnection limit (forced-pushed)

File: src/daemon.rs [hunk 1]
```diff
@@ -129,8 +132,13 @@ struct Connection {
 
 fn tcp_connect(addr: SocketAddr, signal: &Waiter) -> Result<TcpStream> {
     loop {
-        match TcpStream::connect(addr) {
-            Ok(conn) => return Ok(conn),
+        match TcpStream::connect_timeout(&addr, DAEMON_TIMEOUT) {
+            Ok(conn) => {
+                // can only fail if DAEMON_TIMEOUT is 0
+                conn.set_read_timeout(Some(DAEMON_TIMEOUT)).unwrap();
```
Comment (philippem) [inline src/daemon.rs]: we need to have separate timeouts for connection, read, and write.
Connections should be quick, but read and writes may be slow --- for example, a large batch request is made from electrs, and the chain daemon takes many seconds to respond.

**Comment by philippem** [comment]
squash?

**Review by philippem — APPROVED** [review]

**Pull request merged**